src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
How about `strings_matching_regex`? It's a bit verbose, but it covers both unicode and byte strings, is pretty hard to misunderstand, and I feel fits the existing names better.

(We're hella bikeshedding this, but Hypothesis has a legacy of names that weren't bikeshed enough and now we're stuck with, and it would be nice to suffer less of that in future)